src/base-app/venue.cpp

```diff
@@ -22,15 +34,27 @@ namespace base {
 
     //setters
     bool venue::setAddress(QString a){
+        if (a == NULL)
+            return false;
+        address = a;
         return true;
     }
     bool venue::setPhoneNumber(QString n){
+        if (n == NULL)
+            return false;
+        phoneNumber = n;
         return true;
     }
     bool venue::setName(QString n){
+        if (n = NULL)
+            return false;
+        name = n;
         return true;
     }
     bool venue::setEntryFee(double f){
+        if (f == NULL)

```

Comment:
`f == 0` should probably be allowed. I think you just missed these NULLS towards the end when you were making the changes.